### PR TITLE
Radio Page

### PR DESCRIPTION
**Pull request by drigato**

### CODE REVIEW

File: airtime_mvc/application/Bootstrap.php [hunk 1]
```diff
@@ -133,7 +133,11 @@ protected function _initHeadLink()
         $view->headLink()->appendStylesheet($baseUrl . 'css/redmond/jquery-ui-1.8.8.custom.css?' . $CC_CONFIG['airtime_version']);
         $view->headLink()->appendStylesheet($baseUrl . 'css/pro_dropdown_3.css?' . $CC_CONFIG['airtime_version']);
         $view->headLink()->appendStylesheet($baseUrl . 'css/qtip/jquery.qtip.min.css?' . $CC_CONFIG['airtime_version']);
-        $view->headLink()->appendStylesheet($baseUrl . 'css/styles.css?' . $CC_CONFIG['airtime_version']);
+
+        // Don't include style.css on the Radio Page because there are conflicting styles.
+        if ($_SERVER['REQUEST_URI'] != "/") {
+            $view->headLink()->appendStylesheet($baseUrl . 'css/styles.css?' . $CC_CONFIG['airtime_version']);
+        }
```
Comment (asantoni) [inline airtime_mvc/application/Bootstrap.php]: This is a nasty hack. Is there not a better way to remove this explicitly from one page? 

In the long-term, these stylesheets should not even be included in the Bootstrap file. We should have some sort appendDefaultStylesheets() function that's injected in our controllers.

File: airtime_mvc/application/common/WidgetHelper.php [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    // Second version of this function.
+    // Removing "next" days and creating two weekly arrays
+    public static function getWeekInfoV2($timezone)
+    {
+        //weekStart is in station time.
+        $weekStartDateTime = Application_Common_DateHelper::getWeekStartDateTime();
+
+        $dow = array("monday", "tuesday", "wednesday", "thursday", "friday",
+            "saturday", "sunday");
```
Comment (asantoni) [inline airtime_mvc/application/common/WidgetHelper.php]: A) These hardcoded day of week constants could be generated with date().
B) The API is not for human consumption, so outputting an array of labelled days seems weird. They don't need explicit labels if we're always returning Monday to Sunday. It could just be an array of 7 dictionaries.
C) I think we should consider always returning 7 days from the current day. If it's a Sunday, what good does showing me the past 6 days do?

File: airtime_mvc/application/common/WidgetHelper.php [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        if (array_key_exists($userDefinedTimezone, timezone_abbreviations_list())) {
+            $timezone = $userDefinedTimezone;
+        }
+        $utcTimezone = new DateTimeZone("UTC");
+
+        $weekStartDateTime->setTimezone($utcTimezone);
+        $utcDayStart = $weekStartDateTime->format("Y-m-d H:i:s");
+        $weekCounter = 0;
+        while ($weekCounter < $maxNumOFWeeks) {
+            for ($i = 0; $i < 7; $i++) {
```
Comment (asantoni) [inline airtime_mvc/application/common/WidgetHelper.php]: Make the 7 a constant, like DAYS_PER_WEEK. Also try to avoid the use of $i and make your iterables more readable. If $i is the day of the week, just call it $dayOfWeek. It helps with readability in longer for-loops like this.

File: airtime_mvc/application/controllers/EmbedController.php [hunk 4]
```diff
@@ -54,4 +63,43 @@ public function playerAction()
...
+
+        $widgetStyle = $request->getParam('style');
+        if ($widgetStyle == "premium") {
+            $this->view->widgetStyle = "premium";
+            $this->view->css = Application_Common_HTTPHelper::getStationUrl() . "/css/embed/weekly-schedule-widget.css?" . $CC_CONFIG['airtime_version'];
+        } else {
+            $this->view->widgetStyle = "basic";
+            $this->view->css = Application_Common_HTTPHelper::getStationUrl() . "/css/embed/weekly-schedule-widget-basic.css?" . $CC_CONFIG['airtime_version'];
+        }
+        $this->view->jquery = Application_Common_HTTPHelper::getStationUrl() . "widgets/js/jquery-1.6.1.min.js?".$CC_CONFIG['airtime_version'];
```
Comment (asantoni) [inline airtime_mvc/application/controllers/EmbedController.php:95]: This is another thing we should fix at some point. Hardcoding jquery seems bad. Would be good if we had a factory for all these includes...

File: airtime_mvc/application/controllers/EmbedController.php [hunk 5]
```diff
@@ -54,4 +63,43 @@ public function playerAction()
...
+
+        $result = WidgetHelper::getWeekInfoV2($this->getRequest()->getParam("timezone"));
+        $this->view->scheduleDataWeek1 = $result[0];
+        $this->view->scheduleDataWeek2 = $result[1];
```
Comment (asantoni) [inline airtime_mvc/application/controllers/EmbedController.php]: Why use two separate variables instead of an array?

File: airtime_mvc/application/common/WidgetHelper.php (hunk 2 above)
Comment (danielhjames) [inline airtime_mvc/application/common/WidgetHelper.php]: Day of the week labels might be useful for debugging API problems, since Airtime allows the user to set 'Week Starts On:' in the Preferences page.
Displaying previous days is only useful in combination with the API exporting a show URL pointing to a page where the show can be downloaded as a file, i.e. as a catch-up service. In that circumstance, users probably want the seven days ahead plus the seven days behind. BBC Radio is now offering up to one month behind for catch-up.

File: airtime_mvc/application/controllers/EmbeddablewidgetsController.php [hunk 6]
```diff
@@ -20,12 +21,17 @@ public function customizeAction()
         $numEnabledStreams = $form->getElement('player_stream_url')->getAttrib('numberOfEnabledStreams');
 
         if ($numEnabledStreams > 0 && $apiEnabled) {
-            $this->view->form = $form;
+            $this->view->player_form = $form;
         } else {
-            $this->view->errorMsg = "To configure and use the embeddable player you must:<br><br>
+            $this->view->player_error_msg = "To configure and use the embeddable player you must:<br><br>
```
Comment (asantoni) [inline airtime_mvc/application/controllers/EmbeddablewidgetsController.php]: This string should be wrapped in _pro()

File: airtime_mvc/application/controllers/EmbeddablewidgetsController.php [hunk 7]
```diff
@@ -20,12 +21,17 @@ public function customizeAction()
...
             1. Enable at least one MP3, AAC, or OGG stream under System -> Streams<br>
             2. Enable the Public Airtime API under System -> Preferences";
         }
 
+        if (!$apiEnabled) {
+            $this->view->weekly_schedule_error_msg = "To use the embeddable weekly schedule you must:<br><br>
```
Comment (asantoni) [inline airtime_mvc/application/controllers/EmbeddablewidgetsController.php]: Same as above, wrap in _pro()

File: airtime_mvc/application/layouts/scripts/radio-page.phtml [hunk 8]
```diff
@@ -0,0 +1,36 @@
+<?php echo $this->doctype() ?>
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link href='http://fonts.googleapis.com/css?family=Roboto:400,100,300,700' rel='stylesheet' type='text/css'>
+
+    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js"></script>
```
Comment (asantoni) [inline airtime_mvc/application/layouts/scripts/radio-page.phtml:8]: Is this not a second copy of jquery??

File: airtime_mvc/application/layouts/scripts/radio-page.phtml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        . $company
+        . "</a>";
+    echo sprintf(_('%1$s copyright &copy; %2$s All rights reserved.<br />'
+        . 'Maintained and distributed under the %3$s by %4$s'),
+        PRODUCT_NAME, $company,
+        $licenseSiteAnchor,
+        $companySiteAnchor);
+    ?>
+</div>
+-->
```
Comment (asantoni) [inline airtime_mvc/application/layouts/scripts/radio-page.phtml:34]: Remove the commented code if you're done

File: airtime_mvc/application/models/Preference.php [hunk 10]
```diff
@@ -498,7 +498,12 @@ public static function SetStationDescription($description)
 
     public static function GetStationDescription()
     {
-        return self::getValue("description");
+        $description = self::getValue("description");
+        if (!empty($description)) {
+            return $description;
+        } else {
+            return "Powered by Airtime Pro";
```
Comment (asantoni) [inline airtime_mvc/application/models/Preference.php:505]: _pro() translation wrapper? (Maybe? This is a weird case...)

File: airtime_mvc/application/models/Preference.php [hunk 11]
```diff
@@ -602,7 +607,14 @@ public static function SetStationLogo($imagePath)
...
     public static function GetStationLogo()
     {
-        return self::getValue("logoImage");
+        $logoImage = self::getValue("logoImage");
+        if (!empty($logoImage)) {
+            return $logoImage;
+        } else {
+            // We return the Airtime logo if no logo is set in the database.
+            // airtime_logo.png is stored under the public directory
+            return "airtime_logo.png";
```
Comment (asantoni) [inline airtime_mvc/application/models/Preference.php:616]: This is misleading because in airtime_mvc/application/views/scripts/form/preferences_general.phtml you're actually setting the image source to "airtime_logo.png" conditionally. (You've repeated yourself here...)

By reading the code above, I would expect that changing the "airtime_logo.png" string to "foobar.png" would make the default logo point to "foobar.png", but the code in preferences_general.phtml doesn't work like that.

I didn't realize that this function had to return a base64-encoded image. In this case, I think it'd be better to return a magic constant indicating that the default station logo should be used, and then also set a constant for that. (See my comments below in preferences_general.phtml for clarification).

Specifically, on line 616 above I'd return DEFAULT_LOGO_PLACEHOLDER, and define that in constants.php.

File: airtime_mvc/application/views/scripts/form/preferences_general.phtml [hunk 12]
```diff
@@ -12,7 +12,15 @@
 <!--        <button id="logo-remove-btn" value="Remove" class="btn">Remove</button>-->
 
         <div id="Logo-img-container">
-            <img onError="this.onerror = '';this.style.visibility='hidden';$('#logo-remove-btn').hide();$('[id^=stationLogoRemove]').each(function(i,v){v.style.width=v.style.height=v.style.margin=v.style.padding='0px';});" id="logo-img" onload='resizeImg(this, 450, 450);' src="data:image/png;base64,<?php echo $this->element->getView()->logoImg ?>" />
+            <?php
+            $logoImg = $this->element->getView()->logoImg;
+            if ($logoImg === "airtime_logo.png") {
+                $src = "airtime_logo.png";
```
Comment (asantoni) [inline airtime_mvc/application/views/scripts/form/preferences_general.phtml:18]: Continuing my comment from above... I'd change this logic to be like, if ($logoImg === DEFAULT_LOGO_PLACEHOLDER) then $src = DEFAULT_LOGO_FILE.

where DEFAULT_LOGO_FILE would be a constant defined to "airtime_logo.png" in constants.php.

File: airtime_mvc/application/models/Preference.php (hunk 10 above)
Comment (drigato) [inline airtime_mvc/application/models/Preference.php:505]: Maybe we should only translate the "Powered by" part?

Comment (asantoni) [inline airtime_mvc/application/models/Preference.php:505]: Yeah. "Airtime Pro" should be one of the branding constants too, with that interpolated into it with sprintf.

Comment (danielhjames) [inline airtime_mvc/application/models/Preference.php:505]: From a localization point of view I would suggest translating the whole string, because in some languages the word order might be: 'Airtime Pro powered by' even if you do not translate 'Airtime Pro' itself. You might also want the flexibility to insert another brand name here for a white-label client, by changing only the .po file.

Comment (asantoni) [inline airtime_mvc/application/models/Preference.php:505]: This is what sprintf is for. You put the % placeholder in the translatable string and that allows the translator to place the "Airtime Pro" string correctly according to their grammar.

Comment (danielhjames) [inline airtime_mvc/application/models/Preference.php:505]: Using a placeholder is better than nothing, although then the translator might need to be provided with some context. For example 'Powered by %' in the case of Airtime Pro is an idiom and might be difficult to translate, compared to say 'Powered by % lithium-ion batteries' where % is the brand of batteries.

Comment (asantoni) [inline airtime_mvc/application/models/Preference.php:505]: Good translation frameworks give you the option to provide context or
disambiguation in your translation macros. I believe gettext does this if
you add the pipe symbol  | to your strings. We may need to be running
gettext with some extra argument as well.

On Wed, Jun 10, 2015 at 12:39 PM, Daniel James notifications@github.com
wrote:

> In airtime_mvc/application/models/Preference.php
> https://github.com/sourcefabric/airtime/pull/116#discussion_r32137799:
> 
> > @@ -498,7 +498,12 @@ public static function SetStationDescription($description)
> > 
> > ```
> >  public static function GetStationDescription()
> >  {
> > ```
> > -        return self::getValue("description");
> > -        $description = self::getValue("description");
> > -        if (!empty($description)) {
> > -            return $description;
> > -        } else {
> > -            return "Powered by Airtime Pro";
> 
> Using a placeholder is better than nothing, although then the translator
> might need to be provided with some context. For example 'Powered by %' in
> the case of Airtime Pro is an idiom and might be difficult to translate,
> compared to say 'Powered by % lithium-ion batteries' where % is the brand
> of batteries.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/sourcefabric/airtime/pull/116/files#r32137799.

## 

Albert Santoni
Airtime Lead Developer, Sourcefabric
albert.santoni@sourcefabric.org

www.sourcefabric.org

720 Bathurst St.
Suite 203
Toronto, ON / M5S 2R4 / Canada

Comment (danielhjames) [inline airtime_mvc/application/models/Preference.php:505]: https://core.trac.wordpress.org/ticket/9112 describes both context and comment lines in .po files. We would need to test if these show up in the Transifex interface.
